core/src/test/java/com/linecorp/armeria/common/http/MulticastHttpResponseFactoryTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.http;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Test;
+
+public class MulticastHttpResponseFactoryTest {
+
+    @Test
+    public void aggregateTwice() throws ExecutionException, InterruptedException {
+        final DefaultHttpResponse publisher = new DefaultHttpResponse();
+        final MulticastHttpResponseFactory factory = new MulticastHttpResponseFactory(publisher);
+
+        publisher.write(HttpHeaders.of(HttpStatus.OK).set(HttpHeaderNames.CONTENT_TYPE, "text/plain"));
+        publisher.write(HttpData.ofUtf8("Armeria "));
+        publisher.write(HttpData.ofUtf8("is "));
+        publisher.write(HttpData.ofUtf8("awesome!"));
+        publisher.close();
+
+        final AggregatedHttpMessage res1 = factory.newChildStream().aggregate().get();
+        final AggregatedHttpMessage res2 = factory.newChildStream().aggregate().get();
```

Comment:
Global comment: You could use `join()` instead of `get()` to simplify the `throws` clause of the test methods.